server/src/main/java/org/elasticsearch/transport/TransportHandshaker.java

```diff
@@ -203,24 +210,33 @@ public void writeTo(StreamOutput streamOutput) throws IOException {
     static final class HandshakeResponse extends TransportResponse {
 
         private final Version responseVersion;
+        private final Version responseMinimalCompatibleVersion;
 
         HandshakeResponse(Version version) {
             this.responseVersion = version;
+            this.responseMinimalCompatibleVersion = version.minimumCompatibilityVersion();
         }
 
-        private HandshakeResponse(StreamInput in) throws IOException {
-            responseVersion = Version.readVersion(in);
+        @VisibleForTesting
+        HandshakeResponse(StreamInput in) throws IOException {
+            responseVersion = in.getVersion();
+            responseMinimalCompatibleVersion = Version.readVersion(in);
```

Comment:
Does this now read more than before? If so - does that work because exceptions are handled in all cases, or was this just moved?

Reply:
Yes as no streaming change did happen. The `responseVersion` is taken from the streams header which was already always transmitted. The 2nd version is read/written from/to the message body (the request), this was already the case before.

Reply:
I have the same question, and is it possible to test this? (have a sender that doesn't write the second version)

Reply:
haven't seen the new comment before posting, sorry.